src/Silex/Application.php

```diff
@@ -64,8 +63,12 @@ public function __construct(array $values = array())
 
         $app = $this;
 
-        $this['routes'] = function () {
-            return new RouteCollection();
+        $this['routes_class'] = 'Symfony\\Component\\Routing\\RouteCollection';
+        $this['routes_factory'] = $this->factory(function () use ($app) {
```

Comment:
Thanks for considering this PR and for the feedback! I'll look into it and update the PR soon.